src/types.rs

```diff
@@ -550,6 +553,7 @@ impl SourceMap {
         names: Vec<String>,
         sources: Vec<String>,
         sources_content: Option<Vec<Option<String>>>,
+        debug_id: Option<DebugId>,
```

Comment:
However this function I believe is public, so this would be a breaking change.

Reply:
We can leave out `debug_id` from here and set it inside decoder, just as we do with `sm.set_source_root(rsm.source_root);`.

Reply:
Fair point. There's also a setter, so this doesn't need to be passed in `new`.